crypto/compactcert/verifier.go

```diff
@@ -48,28 +48,33 @@ func (v *Verifier) Verify(c *Cert) error {
 		return fmt.Errorf("cert signed weight %d <= proven weight %d", c.SignedWeight, v.ProvenWeight)
 	}
 
-	hsh, err := crypto.HashFactory{HashType: crypto.Sha512_256}.NewHash()
+	// Verify all of the reveals
+	sHash, err := c.SigProofs.HashFactory.NewHash()
+	if err != nil {
+		return err
+	}
+
+	pHash, err := c.PartProofs.HashFactory.NewHash()
```

Comment:
is it possible to extract the correct hash function inside the verification of the tree